### PR TITLE
Arc Shape improvements

### PR DESCRIPTION
**Pull request by kingadami**
This adds full support for width to the pyglet.shape.Arc class. Closes issue #1026

### CODE REVIEW

File: pyglet/shapes.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ def on_draw():
 from abc import ABC, abstractmethod
 
 import pyglet
+import time
```
Comment (benmoran56) [inline pyglet/shapes.py:52]: Unused import left over from testing?

File: pyglet/shapes.py [hunk 2]
```diff
@@ -555,15 +561,109 @@ def _get_vertices(self):
...
+                v3 = (p1[0] - miter1ScaledP[0], p1[1] - miter1ScaledP[1])
+                v4 = (p2[0] + miter2ScaledP[0], p2[1] + miter2ScaledP[1])
+                v5 = (p2[0] - miter2ScaledP[0], p2[1] - miter2ScaledP[1])
+                v6 = (p1[0] - miter1ScaledP[0], p1[1] - miter1ScaledP[1])
+
+                # Setup for next run of function
+                prev_miter = v_miter2
+                prev_scale = scale2
+
+                return (v1[0],v1[1],v2[0],v2[1],v3[0],v3[1],v4[0],v4[1],v5[0],v5[1],v6[0],v6[1])
```
Comment (benmoran56) [inline pyglet/shapes.py:636]: Could you add spaces between the items to keep the linters happy?

**Comment by benmoran56** [comment]
I made a few inline comments, but overall looks good! 

One thing I was thinking about is the usage of `width`.  I ran into a case with the `Box` shape (made up of four lines), which already uses `width` to specify the overall width of the object. I instead opted to use `thickness` for the line widths. What do you think about using that here as well, for consistency. Your call on that.

File: pyglet/shapes.py (hunk 1 above)
Comment (kingadami) [reply pyglet/shapes.py:52]: Yes it was and is now removed

**Comment by kingadami** [comment]
> How about `pyglet.shape.BezierCurve`? It's similar to the Arc.

I am definitely planning on doing that as well.  Since this was my first pull request I was wanting to keep it fairly contained and small so I can get feedback on style and preference.  If this gets accepted then I plan on changing BezierCurve to use this method as well.  I will probably move the _get_segments function outside of the Arc class since the code will be basically the same.

I was also thinking about either extending the `pyglet.shape.Line` to allow multiple points or adding something like a `pyglet.shape.PolyLine` and then re-writing the `pyglet.shape.Line` class as a specialization of the `pyglet.shape.PolyLine` class or something like that.  It would use the same _get_segment function as well, assuming people are open to the idea.

File: pyglet/shapes.py (hunk 2 above)
Comment (kingadami) [reply pyglet/shapes.py]: Done

**Review by benmoran56 — APPROVED** [review]

**Comment by benmoran56** [comment]
That's a solid idea. The `_get_segment` code can easily be a function at the top of the file, and just take in prev_miter/scale as arguments.

**Pull request merged**